lib/modules/dosomething/dosomething_northstar/dosomething_northstar.module

```diff
@@ -20,14 +20,17 @@ define('NORTHSTAR_URL', variable_get('dosomething_northstar_url', 'http://norths
  * @return DoSomething\Gateway\Northstar
  */
 function dosomething_northstar_client() {
+  $oauth_client = variable_get('openid_connect_client_northstar',
+    array('client_id' => NULL, 'client_secret' => NULL));
   return new Northstar([
-    'grant' => 'client_credentials',
+    'grant' => 'authorization_code',
     'url' => NORTHSTAR_URL,
     'bridge' => PhoenixOAuthBridge::class,
-    'client_credentials' => [
```

Comment:
We should keep this config array in case we do want to make requests on behalf of the client later on.

```diff
     'grant' => 'authorization_code',
     'url' => NORTHSTAR_URL,
     'bridge' => PhoenixOAuthBridge::class,
+    'client_credentials' => [
+      'client_id' => variable_get('dosomething_northstar_app_id', 'trusted-test-client'),
+      'client_secret' => variable_get('dosomething_northstar_app_key', 'secret1'),
+      'scope' => ['admin', 'user'],
+    ],
     'authorization_code' => [
       'client_id' => $oauth_client['client_id'],
       'client_secret' => $oauth_client['client_secret'],
```